worker/provisioner/provisioner_task.go

```diff
@@ -1236,17 +1233,25 @@ func (task *provisionerTask) startMachine(
 
 // gatherCharmLXDProfiles consumes the charms LXD Profiles from the different
 // sources. This includes getting the information from the broker.
-func (task *provisionerTask) gatherCharmLXDProfiles(instanceId, machineTag string, machineProfiles []string) []string {
-	if names.IsContainerMachine(machineTag) {
-		if manager, ok := task.broker.(container.LXDProfileNameRetriever); ok {
-			if profileNames, err := manager.LXDProfileNames(instanceId); err == nil {
-				return lxdprofile.LXDProfileNames(profileNames)
-			}
-		} else {
-			task.logger.Tracef("failed to gather profile names, broker didn't conform to LXDProfileNameRetriever")
-		}
+func (task *provisionerTask) gatherCharmLXDProfiles(
+	instanceId, machineTag string, machineProfiles []string,
+) ([]string, error) {
+	if !names.IsContainerMachine(machineTag) {
+		return machineProfiles, nil
+	}
+
+	manager, ok := task.broker.(container.LXDProfileNameRetriever)
+	if !ok {
+		task.logger.Tracef("failed to gather profile names; broker does not implement LXDProfileNameRetriever")
+		return machineProfiles, nil
+	}
+
+	profileNames, err := manager.LXDProfileNames(instanceId)
```

Comment:
I can't remember why we didn't use this error, it seems like an oversight!?

:+1: